generator/README.md

```diff
@@ -10,17 +10,30 @@ Updating the generated code can be done only by modifying the code generator, or
 To run the generator, you need to have PHP 8.1+ installed and Composer.
 
 1. Move to the `generator` directory: `cd generator`
-1. Install dependencies: `composer install`
-1. Run the generator: `./generate`
+2. Install dependencies: `composer install`
+3. Run the generator: `./generate`
 
 ## Configuration
 
 The `generator/config/*.yaml` files contains the list of operators and stages that are supported by the library.
 
+### Arguments
+
+| Field | Type | Description |
+| `name` | `string` | The name of the argument. It can start with `$`. |
+| `type` | list of `string` | The list of accepted types |
+| `description` | `string` | The description of the argument from MongoDB's documentation. |
+| `optional` | `boolean` | Whether the argument is optional or not. |
+| `valueMin` | `number` | The minimum value for a numeric argument. |
+| `valueMax` | `number` | The maximum value for a numeric argument. |
+| `variadic` | `string` | If sent, the argument is variadic. Defines the format `array` for a list or `object` for a map |
+| `variadicMin` | `integer` | The minimum number of arguments for a variadic parameter. |
+| `default` | `scalar` or `array` | The default value for the argument. |
+| `noName` | `bool` | Default `false`. If `true`, the value must be an object and the properties of the value object are merged into the parent operator. `$group` stage uses it for the fields. |
```

Comment:
I'll let @jmikola add his thoughts as well, but "pulling to the top" is usually called "hoisting". An example of this would be [JavaScript Hoisting](https://www.w3schools.com/js/js_hoisting.asp), where variable declarations are hoisted to the top of the block they appear in. Both `noName` and `hoist` would need explanations here, but I feel like `noName` is more ambiguous than `hoist`.

A different alternative is `mergeToTopLevel`, referring to the "merge" wording you used in the field description in the JSON schema.

Reply:
`hoist` is too unfamiliar to me. I'll change to `mergeObject`.

Reply:
I'm fine with `mergeObject` or `mergeUp`. I read the "JavaScript Hoisting" article, putting aside my historic disdain for [W3Fools](https://www.w3fools.com/), but it didn't seem entirely relevant as it was talking about variable declarations and scopes.